### PR TITLE
Ammended version of PR-50

### PR DESCRIPTION
**Pull request by adamretter**
Closes https://github.com/weXsol/Mongrel/pull/50

@dizzzz This doesn't get us 100% of the way there, but things are now much better. Compilation and JUnit steps complete. However no actual XQSuite tests are executed (for the time being you also need to make a copy of `xquery-test-runner.xq` from eXist-db to `test/src/org/exist/test/runner/test-runner.xq` due to a badly coded path issue)...

I might need to make some small changes to XSuite to make it more flexible and can then release an eXist-db 4.1.1 for you.

### CODE REVIEW

File: src/test/resources/org/exist/mongodb/test/db/map-reduce.xqm [hunk 1]
```diff
@@ -3,14 +3,13 @@ xquery version "3.1";
 module namespace mapreduce="http://exist-db.org/mongodb/test/mapreduce";
 
 
-import module namespace test="http://exist-db.org/xquery/xqsuite" 
-                at "resource:org/exist/xquery/lib/xqsuite/xqsuite.xql";
+declare namespace test="http://exist-db.org/org.exist.mongodb.test/xqsuite";
 
 import module namespace mongodb = "http://expath.org/ns/mongo" 
-                at "java:org.exist.mongodb.xquery.MongodbModule";
+                at "java:org.exist.mongodb.org.exist.mongodb.test.MongodbModule";
```
[the remark this answers is not in the document]
Comment (adamretter) [reply src/test/resources/org/exist/mongodb/test/db/map-reduce.xqm]: Ah. I will correct

**Comment by dizzzz** [comment]
this is interesting:

```
(Line: 57 /home/travis/build/weXsol/Mongrel/src/test/resources/org/exist/mongodb/test/db/group.xqm) 
05:51:12.375 [main] ERROR org.exist.xquery.util.DocUtils - java.net.URISyntaxException: xmldb URI scheme does not start with xmldb:: resource:org/exist/mongodb/test/db
```

**Comment by dizzzz** [comment]
I'll rework the test later today/tomorrow.....

**Comment by dizzzz** [comment]
I;ll use this as basis for continuing development. Many thanks !

**Pull request merged**